src/main/java/appeng/crafting/CraftingTreeProcess.java

```diff
@@ -241,8 +241,9 @@ IAEItemStack getAmountCrafted(IAEItemStack what2) {
             final boolean perfectMatch = is.getItem() == what2.getItem()
                     && (is.getItem().isDamageable() || is.getItemDamage() == what2.getItemDamage());
             if (perfectMatch
-                    || (this.details.canSubstitute() || this.details.canBeSubstitute())
-                            && is.fuzzyComparison(what2, FuzzyMode.IGNORE_ALL)) {
+                    || ((this.details != null)
```

Comment:
This probably should be `this.details.canBeSubstitute()` only instead of a OR. `this.details.canSubstitute()` being true will not influence whether this is returned by getCraftingFor.